Common/Securities/SecurityDefinitionSymbolResolver.cs

```diff
@@ -0,0 +1,163 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using QuantConnect.Configuration;
+using QuantConnect.Data.Auxiliary;
+using QuantConnect.Interfaces;
+using QuantConnect.Logging;
+using QuantConnect.Util;
+
+namespace QuantConnect.Securities
+{
+    /// <summary>
+    /// Resolves standardized security definitions such as FIGI, CUSIP, ISIN, SEDOL into
+    /// a properly mapped Lean <see cref="Symbol"/>.
+    /// </summary>
+    public class SecurityDefinitionSymbolResolver
+    {
+        private readonly List<SecurityDefinition> _securityDefinitions;
+        private readonly MapFileResolver _mapFileResolver;
+        
+        /// <summary>
+        /// Creates an instance of the symbol resolver
+        /// </summary>
+        /// <param name="dataFolder">Data folder to read the security-database.csv file from</param>
+        /// <param name="mapFileProvider">Map file provider used to obtain symbol mappings</param>
+        /// <param name="market">The market of the assets referred to by the security definitions</param>
+        public SecurityDefinitionSymbolResolver(
+            DirectoryInfo dataFolder = null,
+            IMapFileProvider mapFileProvider = null,
+            string market = Market.USA)
+        {
+            var dataFolderPath = dataFolder?.FullName ?? Globals.DataFolder;
+            
+            var securityDefinitionFile = new FileInfo(
+                Path.Combine(
+                    dataFolderPath,
+                    "symbol-properties",
+                    "security-database.csv"));            
+            
+            if (!SecurityDefinition.TryFromCsvFile(securityDefinitionFile, out _securityDefinitions))
+            {
+                Log.Error($"SecurityDefinitionSymbolResolver(): No security definitions data loaded from file: {securityDefinitionFile.FullName}");
+            }
+
+            if (mapFileProvider == null)
+            {
+                var dataProvider = Composer.Instance.GetExportedValueByTypeName<IDataProvider>(Config.Get("data-provider", "QuantConnect.Lean.Engine.DataFeeds.DefaultDataProvider"));
+                mapFileProvider = Composer.Instance.GetExportedValueByTypeName<IMapFileProvider>(Config.Get("map-file-provider", "LocalDiskMapFileProvider"));
+                mapFileProvider.Initialize(dataProvider);
+            }
+            
+            _mapFileResolver = mapFileProvider.Get(market);
+        }
+        
+        /// <summary>
+        /// Converts CUSIP into a Lean <see cref="Symbol"/>
+        /// </summary>
+        /// <param name="cusip">CUSIP</param>
+        /// <param name="tradingDate">
+        /// The date that the stock was trading at with the CUSIP provided. This is used
+        /// to get the ticker of the symbol on this date.
```

Comment:
> This is used to get the ticker of the symbol on this date.

Ah nice this is what's missing in the other docs but here `"cusip"` can be improved - Suggest just copy pasting the exact complete docs 👍